makefile

```diff
@@ -2,7 +2,63 @@ buildDir := build
 srcFiles := $(shell find . -name "*.go" -not -path "./$(buildDir)/*" -not -name "*_test.go" -not -path "*\#*")
 testFiles := $(shell find . -name "*.go" -not -path "./$(buildDir)/*" -not -path "*\#*")
 
-_testPackages := ./ 
+packages := pail
+#
+# override the go binary path if set
+ifneq (,$(GO_BIN_PATH))
+gobin := $(GO_BIN_PATH)
+else
+gobin := go
+endif
+
+
+# start linting configuration
+#   package, testing, and linter dependencies specified
+#   separately. This is a temporary solution: eventually we should
+#   vendorize all of these dependencies.
+lintDeps := github.com/alecthomas/gometalinter
+#   include test files and give linters 40s to run to avoid timeouts
+lintArgs := --tests --deadline=13m --vendor
+#   gotype produces false positives because it reads .a files which
+#   are rarely up to date.
+lintArgs += --disable="gotype" --disable="gosec" --disable="gocyclo" --disable="golint"
+lintArgs += --disable="megacheck" --enable="unused" --enable="gosimple"
```

Comment:
we can probably enable golint and megacheck. it's ok if the lint fails at first.